src/Ractive/prototype/set.js

```diff
@@ -10,9 +10,9 @@ export default function Ractive$set ( keypath, value ) {
 	if ( isObject( keypath ) ) {
 		const map = keypath;
 
-		for ( keypath in map ) {
-			if ( map.hasOwnProperty( keypath) ) {
-				set( this, keypath, map[ keypath ] );
+		for ( const k in map ) {
+			if ( map.hasOwnProperty( k) ) {
+				set( this, k, map[k] );
 			}
```

Comment:
@MartinKolarik it does make a huge difference for me. if I count how many times that function is called:

```js
window.foreach = 0
Array.prototype.forEach = function (fn) {
  window.foreach++
  for (var i = 0; i < this.length; i++) {
    fn(this[i], i)
  }
}
```

and then look at `window.foreach` after the page loads I get the value 5460 -- so on our codebase it's called quite a lot. if I specifically record the number of times it's called in that one update function, it's called 192 times for that 20% speed improvement.